File: web/rm-api/src/java/com/rottenmeta/restapi/data/SearchResult.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package com.rottenmeta.restapi.data;

/**
 *
 * @author chloe
 */
public class SearchResult extends RMObject
{

    private Movie movieResult;
    private Critic criticResult;
    private String resultType;
    private int resultLength;

    public SearchResult(String resultType, Object result, int resultLength)
    {
        this.resultType = resultType;
        this.movieResult = resultType.equals("Movie") ? (Movie)result : null;
        this.criticResult = resultType.equals("Critic") ? (Critic)result : null;
        this.resultLength = resultLength;
    }

    public String getResultType() { return this.resultType; }

    public Movie getResultMovie() { return this.movieResult; }
    public Critic getResultCritic() { return this.criticResult; }

    public int getResultLength() { return this.resultLength; }

    @Override
    public String toString()
    {
        return "[SearchResult] (" + resultType + ") " + (movieResult == null ? criticResult == null ? "-" : criticResult : movieResult);
    }

    @Override
    public String getURL()
    {
        return (movieResult == null ? criticResult == null ? "search/" : criticResult.getURL() : movieResult.getURL());
    }
        
}

File: web/rm-web/common/php/object/SearchResult.php
<?php

require_once('Movie.php');
require_once('Critic.php');
require_once('CCObject.php');

class SearchResult extends CCObject
{

    var $result;

    // Accessors
    function getResultType() { return $this->xa['search-result']['_a']['type']; }
    function getResult()
    {
        if(!isset($this->result))
        {
            if($this->getResultType() == "Movie")
                $result = new Movie($this->xa['search-result']['_c']);
            elseif($this->getResultType() == "Critic")
                $result = new Critic($this->xa['search-result']['_c']);
        }
        return $result;
    }

}

?>